compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/expressions/QueryExpressionTest.java

```diff
@@ -281,4 +281,58 @@ public void testQueryActionOrExprWithAsyncSendActionInJoinClause() {
     public void testQueryWithMapKeyword() {
         test("map from int a in b select e", "query-expr/query_expr_assert_79.json");
     }
+
+    @Test
+    public void testQueryWithGroupByUsingVarDefGroupingKeyWithVar() {
+        testFile("query-expr/query_expr_source_80.bal", "query-expr/query_expr_assert_80.json");
+    }
+
+    @Test
+    public void testQueryWithGroupByUsingVarDefGroupingKeyWithTypeDef() {
+        testFile("query-expr/query_expr_source_81.bal", "query-expr/query_expr_assert_81.json");
+    }
+
+    @Test
+    public void testQueryWithGroupByUsingVarNameGroupingKey() {
+        testFile("query-expr/query_expr_source_82.bal", "query-expr/query_expr_assert_82.json");
+    }
+
+    @Test
+    public void testQueryWithGroupByUsingMultipleVarNameGroupingKeys() {
+        testFile("query-expr/query_expr_source_83.bal", "query-expr/query_expr_assert_83.json");
+    }
+
+    @Test
+    public void testQueryWithGroupByUsingMultipleVarDefGroupingKeys() {
+        testFile("query-expr/query_expr_source_84.bal", "query-expr/query_expr_assert_84.json");
+    }
+
+    @Test
+    public void testQueryWithGroupByUsingMultipleGroupingKeys() {
+        testFile("query-expr/query_expr_source_85.bal", "query-expr/query_expr_assert_85.json");
+    }
+
+    @Test
+    public void testQueryWithGroupByRecovery() {
+        test("from int a in b group by select c", "query-expr/query_expr_assert_86.json");
+        test("from int a in b group c select d", "query-expr/query_expr_assert_87.json");
+        test("from int a in b by c select d", "query-expr/query_expr_assert_88.json");
+        test("from int a in b group select c", "query-expr/query_expr_assert_91.json");
+        test("from int a in b by select c", "query-expr/query_expr_assert_92.json");
+        testFile("query-expr/query_expr_source_86.bal", "query-expr/query_expr_assert_89.json");
```

Comment:
We usually sync the source number with the assert file number let's use `query_expr_source_89.bal` for `query_expr_assert_89.json`